js/reference.js

```diff
@@ -717,14 +688,17 @@ var ERMrest = (function(module) {
         get previous() {
             if (this._hasPrevious) {
                 var newReference = _referenceCopy(this._ref);
-                newReference._paging = {};
-                newReference._paging.before = true;
-                newReference._paging.row = {};
-                for (var i = 0; i < newReference._sort.length; i++) {
-                    var col = newReference._sort[i].column;
-                    newReference._paging.row[col] = this._data[0][col]; // first row
-                }
 
+                // update paging by creating a new location
+                var paging = {};
+                paging.before = true;
+                paging.row = [];
+                for (var i = 0; i < newReference._location.sortObject.length; i++) {
+                    var col = newReference._location.sortObject[i].column;
+                    paging.row.push(this._data[0][col]); // first row
+                }
+                // create new _location
+                newReference._location = new module.Location(newReference._location.compactUri + newReference._location.sort + module._getPagingModifier(paging));
```

Comment:
If `Location` had a `clone()` function that returned a new instance of Location based on itself, could this be done like this:

```javascript
var location = this._location.clone();
location.sortObject = ...;
location.pagingObject = ...;

newReference._location = location;
```